encoding/utf8/decode.mbt

```diff
@@ -0,0 +1,179 @@
+#cfg(not(target="js"))
+const U_REP = '\u{FFFD}'
+
+///|
+/// 
+/// References : 
+/// - https://www.unicode.org/versions/Unicode16.0.0/core-spec/chapter-3/#G66453
+/// - https://www.unicode.org/versions/Unicode16.0.0/core-spec/chapter-5/#G40630
+#cfg(not(target="js"))
+pub fn decode_lossy(bytes : @bytes.View) -> String {
+  let builder = StringBuilder::new(size_hint=bytes.length())
+  loop bytes {
+    [] => ()
+    [0..=0x7F as b, .. rest] => {
+      builder.write_char(b.to_int().unsafe_to_char())
+      continue rest
+    }
+    [0xC2..=0xDF as b0, 0x80..=0xBF as b1, .. rest] => {
+      let ch = ((b0.to_int() & 0x1F) << 6) | (b1.to_int() & 0x3F)
+      builder.write_char(ch.unsafe_to_char())
+      continue rest
+    }
+    [0xE0 as b0, 0xA0..=0xBF as b1, 0x80..=0xBF as b2, .. rest]
+    | [0xE1..=0xEC as b0, 0x80..=0xBF as b1, 0x80..=0xBF as b2, .. rest]
+    | [0xED as b0, 0x80..=0x9F as b1, 0x80..=0xBF as b2, .. rest]
+    | [0xEE..=0xEF as b0, 0x80..=0xBF as b1, 0x80..=0xBF as b2, .. rest] => {
+      let ch = ((b0.to_int() & 0x0F) << 12) |
+        ((b1.to_int() & 0x3F) << 6) |
+        (b2.to_int() & 0x3F)
+      builder.write_char(ch.unsafe_to_char())
+      continue rest
+    }
+    [0xE0, 0xA0..=0xBF, .. rest]
+    | [0xE1..=0xEC, 0x80..=0xBF, .. rest]
+    | [0xED, 0x80..=0x9F, .. rest]
+    | [0xEE..=0xEF, 0x80..=0xBF, .. rest] => {
+      builder.write_char(U_REP)
+      continue rest
+    }
+    [
+      0xF0 as b0,
+      0x90..=0xBF as b1,
+      0x80..=0xBF as b2,
+      0x80..=0xBF as b3,
+      .. rest,
+    ]
+    | [
+      0xF1..=0xF3 as b0,
+      0x80..=0xBF as b1,
+      0x80..=0xBF as b2,
+      0x80..=0xBF as b3,
+      .. rest,
+    ]
+    | [
+      0xF4 as b0,
+      0x80..=0x8F as b1,
+      0x80..=0xBF as b2,
+      0x80..=0xBF as b3,
+      .. rest,
+    ] => {
+      let ch = ((b0.to_int() & 0x07) << 18) |
+        ((b1.to_int() & 0x3F) << 12) |
+        ((b2.to_int() & 0x3F) << 6) |
+        (b3.to_int() & 0x3F)
+      builder.write_char(ch.unsafe_to_char())
+      continue rest
+    }
+    [0xF0, 0x90..=0xBF, 0x80..=0xBF, .. rest]
+    | [0xF1..=0xF3, 0x80..=0xBF, 0x80..=0xBF, .. rest]
+    | [0xF4, 0x80..=0x8F, 0x80..=0xBF, .. rest] => {
+      builder.write_char(U_REP)
+      continue rest
+    }
+    [0xF0, 0x90..=0xBF, .. rest]
+    | [0xF1..=0xF3, 0x80..=0xBF, .. rest]
+    | [0xF4, 0x80..=0x8F, .. rest] => {
+      builder.write_char(U_REP)
+      continue rest
+    }
+    [_, .. rest] => {
+      builder.write_char(U_REP)
+      continue rest
+    }
+  }
+  builder.to_string()
+}
+
+///|
+#cfg(target="js")
+pub fn decode_lossy(bytes : @bytes.View) -> String {
+  decode_lossy_js(bytes.data(), bytes.start_offset(), bytes.length())
+}
+
+///|
+#cfg(target="js")
+extern "js" fn decode_lossy_js(
+  bytes : Bytes,
+  start : Int,
+  length : Int,
+) -> String =
+  #|(bytes, start, length) => 
+  #|  new TextDecoder("utf-8", { fatal: false }).decode(new Uint8Array(bytes, start, length))
```

Comment:
TextDecoder if not in Js standard. It's in web standard. Should provide fallback implementation.